core/controllers/collection_editor.py

```diff
@@ -196,6 +196,11 @@ def put(self, collection_id):
                     self.user_id, collection_id)
                 collection_services.index_collections_given_ids([
                     collection_id])
+            elif not is_public:
+                collection_services.unpublish_collection_update_user_profiles(
+                    self.user_id, collection_id)
+                collection_services.index_collections_given_ids([
+                    collection_id])
             elif not self.is_admin:
```

Comment:
This will need to be changed to be ``else:`` since the ``is_admin`` value will be expected to be true in the conditional you added.